src/model_repository_manager.cc

```diff
@@ -2095,38 +2099,36 @@ ModelRepositoryManager::DependencyGraph::Swap(DependencyGraph& rhs)
   std::swap(global_map_ptr_, rhs.global_map_ptr_);
   nodes_.swap(rhs.nodes_);
   missing_nodes_.swap(rhs.missing_nodes_);
+  removed_nodes_.swap(rhs.removed_nodes_);
+}
+
+ModelRepositoryManager::DependencyNode*
+ModelRepositoryManager::DependencyGraph::GetNode(
+    const ModelIdentifier& model_id) const
+{
+  const auto it = removed_nodes_.find(model_id);
+  if (it != removed_nodes_.end()) {
+    return it->second.get();
+  } else {
+    return FindNode(model_id, false /* allow_fuzzy_matching */);
+  }
+  return nullptr;
```

Comment:
```suggestion
  const auto it = removed_nodes_.find(model_id);
  if (it != removed_nodes_.end()) {
    return it->second.get();
  }
  return FindNode(model_id, false /* allow_fuzzy_matching */);
```

Reply:
Nice catch!

Reply:
Updated with the throw exception change.